store/mongo/datastore_mongo.go

```diff
@@ -603,6 +603,49 @@ func (db *DataStoreMongo) AddAuthSet(ctx context.Context, set model.AuthSet) err
 	return nil
 }
 
+// UpsertAuthSetStatus inserts a new auth set and if it exists, ensures the
+// AuthSet.Status matches the one in authSet.
+func (db *DataStoreMongo) UpsertAuthSetStatus(ctx context.Context, authSet *model.AuthSet) error {
+	c := db.client.Database(DbName).Collection(DbAuthSetColl)
+
+	if authSet.Id == "" {
+		uid := oid.NewUUIDv4()
+		authSet.Id = uid.String()
+	}
+	id := identity.FromContext(ctx)
+	tenantId := ""
+	if id != nil {
+		tenantId = id.Tenant
+	}
+	authSet.TenantID = tenantId
+	if authSet.Timestamp == nil {
+		ts := time.Now()
+		authSet.Timestamp = &ts
+	}
+
+	fltr := bson.D{
+		{Key: dbFieldTenantID, Value: tenantId},
+		{Key: dbFieldIDDataSha, Value: authSet.IdDataSha256},
+		{Key: dbFieldPubKey, Value: authSet.PubKey},
+	}
```

Comment:
why you have chosen `UpdateOne` over insert with upsert true? in the latter case we could have just added one param to the `AddAuthSet`?

Reply:
I'm not sure I understand. Did you mean `ReplaceOne` with upsert?
Doing a full replace will update the (inserted) `ts` attribute and potentially other immutable data.